src/stepfunctions/steps/sagemaker.py

```diff
@@ -284,20 +286,10 @@ def __init__(self, state_id, model, model_name=None, instance_type=None, tags=No
             instance_type (str, optional): The EC2 instance type to deploy this Model to. For example, 'ml.p2.xlarge'.
             tags (list[dict], optional): `List to tags <https://docs.aws.amazon.com/sagemaker/latest/dg/API_Tag.html>`_ to associate with the resource.
         """
-        if isinstance(model, FrameworkModel):
+        if isinstance(model, Model):
             parameters = model_config(model=model, instance_type=instance_type, role=model.role, image_uri=model.image_uri)
             if model_name:
                 parameters['ModelName'] = model_name
-        elif isinstance(model, Model):
-            parameters = {
-                'ExecutionRoleArn': model.role,
-                'ModelName': model_name or model.name,
-                'PrimaryContainer': {
-                    'Environment': {},
```

Comment:
I opted for this change since I saw it was supported in sagemaker sdk since the issue was opened, but I agree that we should not introduce breaking changes - even more if the issue does not capture the need to.

I'll revert the changes and go for the solution that was proposed in #84 and add test to validate that behaviour is the same

Reply:
>edit: do we have tests that pass in a FrameworkModel and a Model?

We have tests that use a FrameworkModel (ex: [test_training_step_creation_with_framework()](https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/main/tests/unit/test_sagemaker_steps.py#L503) and others that use a Model (ex: [test_training_step_creation_with_model()](https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/main/tests/unit/test_sagemaker_steps.py#L449)), but none that pass both.

Reply:
> but none that pass both.

I didn't mean in a single test (is that even possible). was just verifying that both parts of the control flow are being tested. I would assume that if they already existed, one of them would have broken with the attempt to introduce a breaking change.

if that didn't happen, i think it surfaces a gap in testing and we should use this opportunity to plug it in

Reply:
I added a test in the latest commit that confirms we are consistent
The test will help catch future breaking changes